Mathlib/CategoryTheory/Galois/Basic.lean

```diff
@@ -0,0 +1,149 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.CategoryTheory.FintypeCat
+import Mathlib.CategoryTheory.Limits.Constructions.LimitsOfProductsAndEqualizers
+import Mathlib.CategoryTheory.SingleObj
+
+/-!
+# Definition and basic properties of Galois categories
+
+We define the notion of a Galois category and a fibre functor as in SGA1, following
+the definitions in Lenstras notes (see below for a reference).
+
+## Main definitions
+
+* `PreGaloisCategory` : defining properties of Galois categories not involving a fibre functor
+* `FibreFunctor`      : a fibre functor from any category to `FintypeCat`
+* `ConnectedObject`   : an object of a category that is not initial and has no non-trivial
+                        subobjects
+
+## Implementation details
+
+We mostly follow Def 3.1 in Lenstras notes. In axiom (G3)
+we omit the factorisation of morphisms in epimorphisms and monomorphisms
+as this is not needed for the proof of the fundamental theorem on Galois categories
+(and then follows from it).
+
+## References
+
+* H. W. Lenstra. Galois theory for schemes
+  <https://websites.math.leidenuniv.nl/algebra/GSchemes.pdf>
+
+-/
+
+universe u₁ u₂ v₁ v₂ w
+
+open CategoryTheory Limits Functor
+
+namespace Galois
+
+section Def
+
+/-- Definition of a Galois category. Lenstra, Def 3.1, (G1)-(G3) -/
+class PreGaloisCategory (C : Type u₁) [Category.{u₂, u₁} C] : Prop where
+  /-- `C` has a terminal object (G1). -/
+  hasTerminalObject : HasTerminal C
```

Comment:
```suggestion
  hasTerminal : HasTerminal C
```

Reply:
Ah, also, for some of the fields, it could be useful to declare them like this
``  hasTerminal : HasTerminal C := by infer_instance``
because then, it will be easier to declare a `PreGaloisCategory` instance, as most of the fields would usually be filled automatically by typeclass inference.